rasa/dialogue_understanding/stack/utils.py

```diff
@@ -106,3 +107,23 @@ def user_flows_on_the_stack(dialogue_stack: DialogueStack) -> Set[str]:
     return {
         f.flow_id for f in dialogue_stack.frames if isinstance(f, UserFlowStackFrame)
     }
+
+
+def end_top_user_flow(stack: DialogueStack) -> None:
+    """Ends all frames on top of the stack including the topmost user frame.
+
+    Ends all flows until the next user flow is reached. This is useful
+    if you want to end all flows that are currently on the stack and
+    the user flow that triggered them.
+
+    Args:
+        stack: The dialogue stack.
+    """
+
+    for frame in reversed(stack.frames):
+        if isinstance(frame, BaseFlowStackFrame):
+            frame.step_id = ContinueFlowStep.continue_step_for_id(END_STEP)
+            if isinstance(frame, UserFlowStackFrame):
+                # Making sure there are no "continue interrupts" triggered
+                frame.frame_type = FlowStackFrameType.REGULAR
```

Comment:
Actually, I suppose in this case where you are simply removing the *top* user flow - a continue interrupt would be a welcome help to let the user know where they are now. I think when we were talking earlier, it was about wrapping up the entire stack, 🤔

Reply:
yeah agreed